app/src/main/java/ghcharactertracker/com/ghcharactertracker/CharacterActivity.java

```diff
@@ -0,0 +1,304 @@
+package ghcharactertracker.com.ghcharactertracker;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.Spinner;
+
+/**
+ * Created by jesse.mailhot on 3/6/2018.
+ */
+
+public class CharacterActivity extends AppCompatActivity {
+    public static final String PLAYER_CHAR = "com.ghcharctertracker.ghcharactertracker.PLAYER_CHAR";
+
+    private static final int SCENARIO_REQUEST_CODE = 0;
+
+    Character character;
+    EditText playerName, playerLevel, playerCurExp, playerMaxHealth, playerNextLvlExp, playerMoney;
+    Spinner classNameSpinner;
+    DBHandler dbHandler;
+
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.character_sheet);
+
+        Intent intent = getIntent();
+        character = (Character) intent.getParcelableExtra(MainActivity.CHARACTER_INPUT);
+        //character = new Character(new CharClass(ClassName.Brute));
+
+        classNameSpinner = (Spinner) findViewById(R.id.classNameSpinner);
+        classNameSpinner.setAdapter(new ArrayAdapter<ClassName>(this,
+                android.R.layout.simple_spinner_item,
+                ClassName.values()));
+
+        playerName = (EditText) findViewById(R.id.playerName);
+        playerLevel = (EditText) findViewById(R.id.playerLevel);
+        playerCurExp = (EditText) findViewById(R.id.playerCurrentExp);
+        playerMaxHealth = (EditText) findViewById(R.id.playerMaxHealth);
+        playerNextLvlExp = (EditText) findViewById(R.id.playerNextLvlExp);
+        playerMoney = (EditText) findViewById(R.id.playerCurrentMoney);
+
+        dbHandler = DBHandler.getDbHandler(this);
+
+        addSpinnerWatcher();
+        addTextWatchers();
+        updateUI();
+    }
+
+    private void updateModelClassName(ClassName className) {
+        if (className != character.getClassName()) {
```

Comment:
Random question that applies to all the next `update` methods.

Is it okay that these can silently do nothing?